tests/protocol/test_connection.py

```diff
@@ -26,3 +26,93 @@ def test_kill(self):
     def test_ping(self):
         self.send_request("ping")
         self.assertEqualResponse("OK")
+
+    def test_tagtypes(self):
+        self.send_request("tagtypes")
+        self.assertInResponse("tagtype: Artist")
+        self.assertInResponse("tagtype: ArtistSort")
+        self.assertInResponse("tagtype: Album")
+        self.assertInResponse("tagtype: AlbumArtist")
+        self.assertInResponse("tagtype: AlbumArtistSort")
+        self.assertInResponse("tagtype: Title")
+        self.assertInResponse("tagtype: Track")
+        self.assertInResponse("tagtype: Name")
+        self.assertInResponse("tagtype: Genre")
+        self.assertInResponse("tagtype: Date")
+        self.assertInResponse("tagtype: Composer")
+        self.assertInResponse("tagtype: Performer")
+        self.assertInResponse("tagtype: Disc")
+        self.assertInResponse("tagtype: MUSICBRAINZ_ARTISTID")
+        self.assertInResponse("tagtype: MUSICBRAINZ_ALBUMID")
+        self.assertInResponse("tagtype: MUSICBRAINZ_ALBUMARTISTID")
+        self.assertInResponse("tagtype: MUSICBRAINZ_TRACKID")
+        self.assertInResponse("OK")
+
+    def test_tagtypes_clear(self):
+        self.send_request("tagtypes clear")
+        self.assertEqualResponse("OK")
+
+    def test_tagtypes_all(self):
+        self.send_request("tagtypes all")
+        self.assertInResponse("tagtype: Artist")
+        self.assertInResponse("tagtype: ArtistSort")
+        self.assertInResponse("tagtype: Album")
+        self.assertInResponse("tagtype: AlbumArtist")
+        self.assertInResponse("tagtype: AlbumArtistSort")
+        self.assertInResponse("tagtype: Title")
+        self.assertInResponse("tagtype: Track")
+        self.assertInResponse("tagtype: Name")
+        self.assertInResponse("tagtype: Genre")
+        self.assertInResponse("tagtype: Date")
+        self.assertInResponse("tagtype: Composer")
+        self.assertInResponse("tagtype: Performer")
+        self.assertInResponse("tagtype: Disc")
+        self.assertInResponse("tagtype: MUSICBRAINZ_ARTISTID")
+        self.assertInResponse("tagtype: MUSICBRAINZ_ALBUMID")
+        self.assertInResponse("tagtype: MUSICBRAINZ_ALBUMARTISTID")
+        self.assertInResponse("tagtype: MUSICBRAINZ_TRACKID")
+        self.assertInResponse("OK")
+
+    def test_tagtypes_disable(self):
+        self.send_request("tagtypes all")
+        self.send_request(
+            "tagtypes disable MUSICBRAINZ_ARTISTID MUSICBRAINZ_ALBUMID "
+            "MUSICBRAINZ_ALBUMARTISTID MUSICBRAINZ_TRACKID"
+        )
+        self.assertInResponse("tagtype: Artist")
```

Comment:
Needs `self.assertNotInResponse("tagtype: MUSICBRAINZ_ARTISTID")` etc else is doesnt actually test the disable function did anything.